2751 blog search query improvements

natalie-bernhard opened the pull request:
#### What was fixed?  (Issue # or description of fix)

Issue #2751
Looks good on draft: https://draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/?search=release&search!=beta

#### Were the changes tested on
- [ ] Firefox (Desktop)
- [ ] Safari (Desktop)
- [ ] Chrome (Desktop)
- [ ] Internet Explorer (Desktop)
- [ ] iOS (Mobile)
- [ ] Android (Mobile)
#### Running validation tools
- [ ] https://validator.w3.org/checklink
- [ ] https://validator.w3.org
- [ ] IBM Equal Access Accessibility Checker (https://www.ibm.com/able/toolkit/verify/automated)
- [ ] Lighthouse (in Chrome dev tools)

kinueng commented:
Hi @natalie-bernhard, is this PR still in draft or may it be marked as `Ready for review`?

natalie-bernhard committed:
bug fixes for exlcuding tags

natalie-bernhard committed:
removing unnecesary code; small css change

natalie-bernhard committed:
draft bug fix merged in

natalie-bernhard committed:
Bug fix to make query order not matter

natalie-bernhard committed:
WIP while waiting for feedback

natalie-bernhard committed:
add simple UI to display adv filters
allow user to see which tags are included or excluded from blog search

natalie-bernhard committed:
enabled multi tag filter & exclude filter in query
Multiple tags and negated tags (=!) can be used in search query
Currently filters based on AND, not OR

natalie-bernhard committed:
Able to exclude tags from filter using url query
Can use !term in blog url search query to omit tags from filter results

natalie-bernhard committed:
small fixes due to rebasing
